packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageBox.tests.js

```diff
@@ -293,6 +297,237 @@ QUnit.module('MessageBox', moduleConfig, () => {
         });
     });
 
+    QUnit.module('onTypingStart event', () => {
+        QUnit.test('should be triggered once if a character is entered in the input', function(assert) {
+            const onTypingStartStub = sinon.stub();
+
+            this.reinit({ onTypingStart: onTypingStartStub });
+
+            keyboardMock(this.$input)
+                .focus()
+                .type('n');
+
+            assert.strictEqual(onTypingStartStub.callCount, 1);
+        });
+
+        QUnit.test('should be possible to update it at runtime', function(assert) {
+            const onTypingStartStub = sinon.stub();
+
+            this.instance.option('onTypingStart', onTypingStartStub);
+
+            keyboardMock(this.$input)
+                .focus()
+                .type('n');
+
+            assert.strictEqual(onTypingStartStub.callCount, 1);
+        });
+
+        ['', ' '].forEach(value => {
+            QUnit.test(`should not be triggered if an empty character is entered in the input, value is '${value}'`, function(assert) {
+                const onTypingStartStub = sinon.stub();
+
+                this.reinit({ onTypingStart: onTypingStartStub });
+
+                keyboardMock(this.$input)
+                    .focus()
+                    .type(value);
+
+                assert.strictEqual(onTypingStartStub.callCount, 0);
+            });
+        });
+
+        QUnit.test('must be called with a delay', function(assert) {
+            const clock = sinon.useFakeTimers({ now: new Date().getTime() });
```

Comment:
Use `test.beforeEach` hook to init `sinon.useFakeTimers()` and will restore it using 'afterEach()' hook

Reply:
Updated